### PR TITLE
Added flush() for S3

### PR DESCRIPTION
**Pull request by nitkul**
Hey Lev, I have addressed issue #113 by learning whatever I could on file flushing.
Hope this helps!

### CODE REVIEW

File: smart_open/smart_open_lib.py [hunk 1]
```diff
@@ -837,6 +844,10 @@ def close(self):
             logger.info("setting an empty value for the key")
             self.outkey.set_contents_from_string('')
 
+    def flush(self):
+        # Implementing as no-op
```
Comment (tmylk) [inline smart_open/smart_open_lib.py]: Please add motivation why. "to match the file interface'

File: smart_open/tests/test_smart_open.py [hunk 2]
```diff
@@ -347,6 +347,18 @@ def test_buffer_flushed_after_eof(self):
...
     @mock_s3
+    def test_open_and_close(self):
+        conn = boto.connect_s3()
+        conn.create_bucket("mybucket")
+        test_string = u"ветер по морю гуляет..."
+        with smart_open.smart_open("s3://mybucket/mykey", "wb") as fout:
+            fout.write(test_string.encode('utf8'))
+        
+        r = smart_open.S3OpenRead(conn.get_bucket("mybucket").get_key("mykey"))
+        r.flush()
```
Comment (tmylk) [inline smart_open/tests/test_smart_open.py]: flush does nothing for readonly streams. what is the motivation for this test?

**Comment by menshikh-iv** [comment]
Ping @nitkul, what's the status of PR?

**Comment by stevendikowitz** [comment]
Hey @nitkul thanks for your work on this! I'm running into a flush issue, and after reviewing the [io docs](https://docs.python.org/3/library/io.html#io.IOBase.flush), it looks like your implementation of flush in this PR is backwards:

`Flush the write buffers of the stream if applicable. This does nothing for read-only and non-blocking streams.`

You have the `pass` method on `S3OpenWrite `, it should be on `S3OpenRead`

**Comment by nitkul** [comment]
hey, @menshikh-iv @tmylk sorry i had to leave work on the PR. might resume it!

@stevendikowitz great! i had a confusion back then but now it's resolved. :-)

**Comment by menshikh-iv** [comment]
Already implemented in current master fbc82cc04bf92e8c588c49c43ff5aff8234ea87d

**Pull request closed without merging**